avail-cli/src/main/kotlin/com/avail/tools/compiler/configuration/CompilerConfiguration.kt

```diff
@@ -75,12 +89,26 @@ class CompilerConfiguration : Configuration
 			var roots = field
 			if (roots === null)
 			{
-				roots = ModuleRoots(availRootsPath)
+				val semaphore = Semaphore(0)
+				roots = ModuleRoots(fileManager, availRootsPath) {
+					it.forEach { msg -> System.err.println(msg) }
+					semaphore.release()
+				}
+				semaphore.acquireUninterruptibly()
 				availRoots = roots
 			}
```

Comment:
This is how I resolved this:

```kotlin
       /**
	 * A [ReentrantLock] for guarding the setting of [availRoots].
	 */
	private val lock = ReentrantLock()

	/** The [Avail roots][ModuleRoots].  */
	private var availRoots: ModuleRoots? = null
		get()
		{
			var roots = field
			// opportunistic check to see if availRoots is already set
			if (roots === null)
			{
				// it isn't set, so try to acquire a lock to set it to avoid
				// other threads trying to set it.
				lock.lock()
				try
				{
					// it is possible we passed into this block while availRoots
					// was unset but was being set by thread that previously
					// acquired the lock. If availRoots is not set, set it now
					if (roots === null)
					{
						val semaphore = Semaphore(0)
						roots = ModuleRoots(fileManager, availRootsPath) {
							it.forEach { msg -> System.err.println(msg) }
							semaphore.release()
						}
						semaphore.acquireUninterruptibly()
						field = roots
					}
				}
				finally
				{
					lock.unlock()
				}
			}
			return roots
		}
```